internal/config/identity/openid/provider/keycloak.go

```diff
@@ -81,9 +81,17 @@ func (k *KeycloakProvider) LoginWithClientID(clientID, clientSecret string) erro
 	return nil
 }
 
+func joinURLPaths(base string, parts ...string) string {
+	res := base
+	for _, part := range parts {
+		res = fmt.Sprintf("%s/%s", strings.TrimRight(res, "/"), strings.TrimLeft(part, "/"))
+	}
+	return res
+}
+
 // LookupUser lookup user by their userid.
 func (k *KeycloakProvider) LookupUser(userid string) (User, error) {
-	lookupUserID := k.adminURL + "/realms" + k.realm + "/users/" + userid
+	lookupUserID := joinURLPaths(k.adminURL, "realms", k.realm, "users", userid)
```

Comment:
This should be rewritten like this instead, joinURLPaths is a complicated piece that is not necessary.

```
k.adminURL + "/" + path.Join("realms", k.realm, "users", userid)
```

Reply:
I understand.... I implemented the helper function since I was not sure if the content of `adminURL` is validated before so that it does not contain a trailing slash. From my understanding (please correct me if I'm wrong), `adminURL` contains the value a user provided via `MINIO_IDENTITY_OPENID_KEYCLOAK_ADMIN_URL` env variable. If that value contains a trailing slash, the composed URL is messed up again. This is why I thought that we could use this little helper function to compose the whole URL (`path.Join` only seams to work for paths and not for fully qualified urls).

Reply:
That really doesn't matter if there are double `//` @hellivan since the HTTP calls filter them out.

Reply:
just change it to `path.Join()` joinURLPaths is a complicated piece that is not safe to use.